### PR TITLE
Fix minor problems in ESLint formatter

### PR DESCRIPTION
**Pull request by lgolding**
- Bring `$schema` up to date.
- Populate `result.ruleIndex` when metadata is available.
- Populate `result.locations[i].physicalLocation.artifactLocation.index`.
- Rule metadata `tags` should be `properties`.
- The appropriate tool component URI here is `informationUri`, not `downloadUri`.

### CODE REVIEW

**Comment by lgolding** [comment]
@michaelcfanning is added to the review. #Closed

**Comment by lgolding** [comment]
@EasyRhinoMSFT is added to the review. #Closed

**Comment by lgolding** [comment]
@boAndron is added to the review. #Closed

**Comment by lgolding** [comment]
@rscrivens is added to the review. #Closed

**Comment by lgolding** [comment]
@boAndron is added to the review. #Closed

**Comment by lgolding** [comment]
@harleenkohli is added to the review. #Closed

File: src/ESLint.Formatter/test/sarif.js [hunk 1]
```diff
@@ -105,14 +121,14 @@ describe("formatter:sarif", () => {
         }];
 
         it("should return a log with one rule", () => {
-            const result = JSON.parse(formatter(code, { rulesMeta: rules }));
```
Comment (harleenkohli) [inline src/ESLint.Formatter/test/sarif.js]: >result [](start = 18, length = 6)

yuck, thanks for fixing the naming here!

**Pull request merged**